assets/thanos-querier/deployment.yaml

```diff
@@ -181,6 +187,8 @@ spec:
           name: secret-thanos-querier-tls
         - mountPath: /etc/kube-rbac-proxy
           name: secret-thanos-querier-kube-rbac-proxy-rules
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```

Comment:
we most likely don't want that

Reply:
We are not platform agnostic. Thanos is built only for linux, same for [prometheus](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/prometheus-k8s/prometheus.yaml#L169-L170), [alertmanager](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/alertmanager/alertmanager.yaml#L106-L107), [ksm](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/kube-state-metrics/deployment.yaml#L105-L106), and all other components:
```console
$ grep nodeSelector -A1 -R . 
./alertmanager/alertmanager.yaml:  nodeSelector:
./alertmanager/alertmanager.yaml-    kubernetes.io/os: linux
--
./grafana/deployment.yaml:      nodeSelector:
./grafana/deployment.yaml-        beta.kubernetes.io/os: linux
--
./kube-state-metrics/deployment.yaml:      nodeSelector:
./kube-state-metrics/deployment.yaml-        kubernetes.io/os: linux
--
./node-exporter/daemonset.yaml:      nodeSelector:
./node-exporter/daemonset.yaml-        kubernetes.io/os: linux
--
./openshift-state-metrics/deployment.yaml:      nodeSelector:
./openshift-state-metrics/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-adapter/deployment.yaml:      nodeSelector:
./prometheus-adapter/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-k8s/prometheus.yaml:  nodeSelector:
./prometheus-k8s/prometheus.yaml-    kubernetes.io/os: linux
--
./prometheus-operator-user-workload/deployment.yaml:      nodeSelector:
./prometheus-operator-user-workload/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-operator/deployment.yaml:      nodeSelector:
./prometheus-operator/deployment.yaml-        kubernetes.io/os: linux
--
./prometheus-user-workload/prometheus.yaml:  nodeSelector:
./prometheus-user-workload/prometheus.yaml-    kubernetes.io/os: linux
--
./telemeter-client/deployment.yaml:      nodeSelector:
./telemeter-client/deployment.yaml-        beta.kubernetes.io/os: linux
--
./thanos-querier/deployment.yaml:      nodeSelector:
./thanos-querier/deployment.yaml-        beta.kubernetes.io/os: linux
```

Reply:
Oh ok, I thought they could be scheduled on any nodes and the respective runtimes would handle the rest.